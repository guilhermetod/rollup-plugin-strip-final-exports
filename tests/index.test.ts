import { readdirSync } from 'fs';
import { join } from 'path';

import { getBundleOutput } from '@tests/helpers/get-bundle-output';
import { specsFolder } from '@tests/helpers/specs-folder';

const cjsFolder = join(specsFolder, 'cjs');
const esmFolder = join(specsFolder, 'esm');

describe('Strip final exports', () => {
  it.each(readdirSync(cjsFolder))('should strip CommonJS final exports from "%s"', async (input) => {
    expect(await getBundleOutput(input, 'cjs')).toMatchSnapshot();
  });

  it.each(readdirSync(esmFolder))('should strip ESModules final exports from "%s"', async (input) => {
    expect(await getBundleOutput(input, 'esm')).toMatchSnapshot();
  });
});
