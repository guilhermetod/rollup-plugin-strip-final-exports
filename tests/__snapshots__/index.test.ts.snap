// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Strip final exports should strip CommonJS final exports from "export-all.cjs" 1`] = `
"'use strict';

var sayHello = function sayHello() {
  return 'Hello, world';
};

var exportAll = sayHello;

"
`;

exports[`Strip final exports should strip CommonJS final exports from "export-as.cjs" 1`] = `
"'use strict';

var sayHello$1 = function sayHello() {
  return 'Hello, world';
};

const { sayHello } = sayHello$1;

var exportAs = { hello: sayHello };

"
`;

exports[`Strip final exports should strip CommonJS final exports from "export-default.cjs" 1`] = `
"'use strict';

var sayHello$1 = function sayHello() {
  return 'Hello, world';
};

const { sayHello } = sayHello$1;

var exportDefault = sayHello;

"
`;

exports[`Strip final exports should strip CommonJS final exports from "named-export.cjs" 1`] = `
"'use strict';

var sayHello$1 = function sayHello() {
  return 'Hello, world';
};

const { sayHello } = sayHello$1;

var namedExport = { sayHello };

"
`;

exports[`Strip final exports should strip ESModules final exports from "export-all.js" 1`] = `
"var sayHello = function sayHello() {
  return 'Hello, world';
};
"
`;

exports[`Strip final exports should strip ESModules final exports from "export-as.js" 1`] = `
"function sayHello() {
  return 'Hello, world';
}

"
`;

exports[`Strip final exports should strip ESModules final exports from "export-default.js" 1`] = `
"function sayHello() {
  return 'Hello, world';
}

"
`;

exports[`Strip final exports should strip ESModules final exports from "named-export.js" 1`] = `
"function sayHello() {
  return 'Hello, world';
}

"
`;
